src/loaders/umx_load.c

```diff
@@ -129,7 +129,8 @@ static int get_objtype (HIO_HANDLE *f, int32 ofs, int type)
 	char sig[16];
 _retry:
 	hio_seek(f, ofs, SEEK_SET);
-	hio_read(sig, 16, 1, f);
+	if (hio_read(sig, 16, 1, f) < 16)
```

Comment:
This is broken: Valid umr files are rejected, because:
- the result should be `1`, not `16`
- do the lines 162 and 174 not need similar checks?
- can we work this around by zeroing the `sig` array?

Reply:
* You're right, missed that part because I'm used to libxmp passing the size into `count` everywhere... I'll revise this.
* 142, 144, 162, 174 probably wouldn't have this issue since the first read is guaranteed to initialize the array.
* That would probably also work and is also guaranteed to initialize the array.

Reply:
>     * 142, 144, 162, 174 probably wouldn't have this issue since the first read is guaranteed to initialize the array.
>     * That would probably also work and is also guaranteed to initialize the array.

You are right, initialization is what only matters - sorry.